File: typeScript/regex.ts
import * as fs from 'fs';

function calculate(pattern: string, text: string): number {
    const start = Date.now();
    const regexPattern = new RegExp(pattern);
    text.match(regexPattern);
    const end = Date.now();
    return end - start;
}

function main() {
    const inputTxtPath = process.argv[2];
    const patternsPath = process.argv[3];
    const resultPath = process.argv[4];

    console.log(`Args - inputTxtPath: ${inputTxtPath}, patternsPath: ${patternsPath}, resultPath: ${resultPath}`);

    if (!inputTxtPath || !patternsPath || !resultPath) {
        console.log('Usage: ts-node main.ts inputTxtPath patternsPath resultPath');
        return;
    }

    const inputTxt = fs.readFileSync(inputTxtPath, 'utf-8');
    const patternsInput = fs.readFileSync(patternsPath, 'utf-8').split('\n');

    const results: string[] = [];
    for (const pattern of patternsInput) {
        let avgTime = 0;
        for (let i = 0; i < 10; i++) {
            const exe = calculate(pattern.trim(), inputTxt);
            avgTime += exe;
        }
        results.push(`${pattern}\t\t${avgTime}`);
    }

    fs.writeFileSync(resultPath, results.join('\n'));
}

main();

File: python/regex.py
import sys
import re
import time

def calculate(pattern, text):
    start = time.time()
    regexPattern = re.compile(pattern)
    regexPattern.findall(text)
    end = time.time()
    elapsed = end - start
    return elapsed * 1000

def main():
    inputTxtPath = sys.argv[1]
    patternsPath = sys.argv[2]
    resultPath = sys.argv[3]

    print("Args - inputTxtPath:", inputTxtPath, "patternsPath:", patternsPath, "resultPath:", resultPath)

    if inputTxtPath == "" or patternsPath == "" or resultPath == "":
        print("Usage: python main.py inputTxtPath patternsPath resultPath")
        return

    with open(inputTxtPath, "r") as inputTxtFile:
        inputTxt = inputTxtFile.read()

    with open(patternsPath, "r") as patternsFile:
        patternsInput = patternsFile.readlines()

    results = []
    for pattern in patternsInput:
        avgTime = 0.0
        for i in range(10):
            exe = calculate(pattern.strip(), inputTxt)
            avgTime += exe
        avgTime /= 10    
        results.append(pattern.strip() + "\t\t" + str(avgTime))

    with open(resultPath, "w") as resultFile:
        resultFile.write("\n".join(results))

if __name__ == "__main__":
    main()
